### PR TITLE
quick change based on new databroker

### PR DESCRIPTION
**Pull request by licode**

### CODE REVIEW

File: suitcase/tests/conftest.py [hunk 1]
```diff
@@ -1,30 +1,43 @@
...
+        'config': {
+            'host': 'localhost',
+            'port': 27017,
+            'database': 'test2'}
+    }
+}
+
+
 @pytest.fixture(params=[1], scope='function')
 def db_all(request):
```
Comment (tacaswell) [inline suitcase/tests/conftest.py]: It is probably better to import the fixtures from databroker here rather than re-writing them.  See what we do in bluesky.

File: suitcase/tests/test_spec.py [hunk 2]
```diff
@@ -188,32 +185,26 @@ def test_round_trip_from_run_engine(db_all):
...
-    sf = spec.Specfile(fname)
-    sf1 = _round_trip(sf, db_all)
+    RE(count([det]))
 
-    # a2scan is not round trippable
-    num_unconvertable_scans = 1
+    RE(outer_product_scan([det], motor, -1, 1, 10, motor1, -1, 1, 10, True))
 
-    assert len(sf) == (len(sf1) + num_unconvertable_scans)
+    with pytest.raises(NotImplementedError):
```
Comment (tacaswell) [inline suitcase/tests/test_spec.py]: ? why is this raising now?

File: suitcase/tests/conftest.py [hunk 3]
```diff
@@ -1,30 +1,43 @@
...
-    db_name = "mds_testing_disposable_{}".format(str(uuid.uuid4()))
-    test_conf = dict(database=db_name, host='localhost',
-                     port=27017, timezone='US/Eastern',
-                     mongo_user='tom',
-                     mongo_pwd='jerry')
-    #version_v = request.param
-    mds = MDS(test_conf, auth=AUTH)
-
-    db = Broker(mds, fs=None)
+    db = Broker.from_config(test_config)
```
Comment (tacaswell) [inline suitcase/tests/conftest.py]: It is better to throw a uid into the database name so that you are sure you get a clean database and are not stepping on anyone's toes.

**Comment by licode** [comment]
If no conflict, this is to be merged soon.

**Comment by tacaswell** [comment]
Can you address why `spec.SpecFile` now errors?

**Comment by licode** [comment]
Ok, work on new comments

File: suitcase/tests/conftest.py (hunk 3 above)
Comment (licode) [reply suitcase/tests/conftest.py]: good point.

File: suitcase/tests/test_spec.py [hunk 4]
```diff
@@ -188,32 +185,26 @@ def test_round_trip_from_run_engine(db_all):
...
     #   suitcase.spec:_get_plan_type
-    RE(ct())
+    RE(count([det]))
 
-    RE(a2scan(motor, -1, 1, motor1, -1, 1, 10))
+    RE(outer_product_scan([det], motor, -1, 1, 10, motor1, -1, 1, 10, True))
 
+    #with pytest.raises(NotImplementedError):
     sf = spec.Specfile(fname)
-    sf1 = _round_trip(sf, db_all)
```
Comment (tacaswell) [inline suitcase/tests/test_spec.py]: why did you delete these tests?

**Comment by tacaswell** [comment]
@danielballan where did you fixes to livespec from CHX end up?

**Comment by danielballan** [comment]
Currently xf11id1-ws1:~xf11id/Repos/suitcase

**Comment by licode** [comment]
Do you want to send a PR for that fix?

**Comment by licode** [comment]
I will take care of that fix.

**Comment by danielballan** [comment]
I appreciate that offer, but I left a mixture of necessary and (we later learned) unnecessary changes. I had better sort through them myself.

**Comment by danielballan** [comment]
I don't think I can do this confidently without testing at CHX one more time. Let's punt this to Tuesday or Wednesday -- depending on how things go at CMS.

**Comment by licode** [comment]
ok, I will review the pr when it is ready. After those fixes, we can add a new version number here.

File: suitcase/tests/test_spec.py (hunk 4 above)
Comment (danielballan) [reply suitcase/tests/test_spec.py]: Bump. This is an important test. Is it deleted because it now fails?

**Comment by tacaswell** [comment]
@licode any progress on fixing the spec related code?

**Comment by danielballan** [comment]
This is waiting on me to extract and review the hotfix I deployed at CHX. Li dutifully reminded me of this -- totally my fault that it hasn't gotten done yet.

File: suitcase/tests/test_spec.py [hunk 5]
```diff
@@ -205,6 +205,12 @@ def test_round_trip_from_run_engine(db_all):
 
     #with pytest.raises(NotImplementedError):
     sf = spec.Specfile(fname)
+    sf1 = _round_trip(sf, db_all)
+
+    # a2scan is not round trippable
```
Comment (danielballan) [inline suitcase/tests/test_spec.py]: Hmm. Why?

Comment (licode) [reply suitcase/tests/test_spec.py]: Old comment, not from me. We may need to update it.

File: suitcase/tests/conftest.py (hunk 3 above)
Comment (danielballan) [reply suitcase/tests/conftest.py]: There are a couple differences here. Notice that the old one created a new, clean database each time. This one would reuse the same database each time, potentially introducing coupling between the tests. Also, the old name was not _just_ a UUID: it started with 'mds_testing_disposable_' so that someone looking at a list of mongo databases has a clue where they came from.

File: suitcase/tests/test_spec.py [hunk 6]
```diff
@@ -183,11 +183,11 @@ def test_round_trip_from_specfile(spec_filename, db_all):
 
 @pytest.mark.skipif(sys.version_info == (2,7),
                     reason="bluesky related tests need python 3.5, 3.6")
-def test_round_trip_from_run_engine(db_all, RE):
-    import bluesky
-    # generate a new specfile
+def test_round_trip_from_run_engine(db_all):
```
Comment (danielballan) [inline suitcase/tests/test_spec.py]: Why not use the `RE` test fixture here?

Comment (licode) [reply suitcase/tests/test_spec.py]: RE is only used in this function. It is easy to make pytest skip this test function for py27. There can be other ways.

File: suitcase/tests/data/temperature_data.py [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+# "Magic numbers" for this simulation
+start, stop, step, points_per_step = 0, 6, 1, 7
+deadband_size = 0.9
+num_exposures = 17
+
+
+def example(func):
+    @wraps(func)
+    def mock_run_start(mds, run_start_uid=None, sleep=0, make_run_stop=True):
```
Comment (danielballan) [inline suitcase/tests/data/temperature_data.py]: As you might remember, all of this stuff -- the whole temperature ramp example -- was written before bluesky even existed. Back then, the only way to compose Documents to test metadatastore/databroker against was to assemble them manually like this. Now, we can do it much more succinct (and correctly) using bluesky. This old code of mine was deleted in favor of that approach, and I'd prefer that it stay gone.

File: suitcase/tests/test_spec.py (hunk 6 above)
Comment (danielballan) [reply suitcase/tests/test_spec.py]: I suspect you will want to use it in other functions, though. See my comment about temperature_ramp.

File: suitcase/tests/data/temperature_data.py (hunk 7 above)
Comment (licode) [reply suitcase/tests/data/temperature_data.py]: good point. it is better not to rewrite example data each time.

**Comment by licode** [comment]
@danielballan , want to quickly check this? I am going to squash commits.

File: suitcase/tests/conftest.py (hunk 3 above)
Comment (danielballan) [reply suitcase/tests/conftest.py]: Do you plan to address this?

Comment (tacaswell) [reply suitcase/tests/conftest.py]: Can you please address this and generate a fresh database every time?

File: suitcase/spec.py [hunk 8]
```diff
@@ -1013,7 +1015,7 @@ def _get_plan_name(start):
             "Until this feature is implemented, we will be using the sequence "
             "number as the motor position".format(plan_name))
         return _NOT_IMPLEMENTED_SCAN
-    return _SPEC_SCAN_NAMES[_BLUESKY_PLAN_NAMES.index(plan_name)]
+    return _BLUESKY_PLAN_NAMES[plan_name]
```
Comment (tacaswell) [inline suitcase/spec.py]: I thought we talked about using `get(plan_name, plan_name)` to let plan names pass without renaming instead of using 'Other'.

File: suitcase/tests/test_hdf5.py [hunk 9]
```diff
@@ -59,87 +62,102 @@ def shallow_header_verify(hdf_path, header, db, fields=None,
...
+                    reason="bluesky related tests need python 3.5, 3.6")
+def test_hdf5_export_single_stream_name(db_all, RE):
     """
     Test the hdf5 export with a single header and
     verify the output is correct. No uid is used.
     """
-    mds = db_all.mds
-    temperature_ramp.run(mds)
+    RE.subscribe(db_all.insert)
+    RE(count([det], 5, delay = 1), owner="Tom")
```
Comment (tacaswell) [inline suitcase/tests/test_hdf5.py]: This is not a like-for-like change.  This should be a `scan` right? 

Same with below.

Comment (licode) [reply suitcase/tests/test_hdf5.py]: I tried scan with motor delays. But timestamps stay the same. Wait for @tacaswell to fix bugs in ophyd, then try again.

Comment (tacaswell) [reply suitcase/tests/test_hdf5.py]: xrif: https://github.com/NSLS-II/ophyd/pull/498

Comment (danielballan) [reply suitcase/tests/test_hdf5.py]: That fix to ophyd has been merged.

**Comment by danielballan** [comment]
This should be a high priority. I think CHX is having problems that are fixed by this PR.

File: suitcase/nexus.py [hunk 10]
```diff
@@ -171,7 +170,7 @@ def export(headers, filename,
                 datetime_string = time.asctime(time.gmtime(start))
                 ds.attrs['start'] = dateutil.parser.parse(datetime_string).isoformat()
 
-                [fill_event(e) for e in events]
+                [db.fill_event(e) for e in events]
```
Comment (danielballan) [inline suitcase/nexus.py]: I believe that `db.fill_events(events)` would work here, and `db.fill_event` is deprecated (if I recall correctly) or at least discouraged.

File: suitcase/tests/conftest.py (hunk 3 above)
Comment (danielballan) [reply suitcase/tests/conftest.py]: Re-upping this.

File: suitcase/spec.py [hunk 11]
```diff
@@ -965,11 +965,13 @@ def to_spec_file_header(start, filepath, baseline_descriptor=None):
 
 _SPEC_1D_COMMAND_TEMPLATE = env.from_string("{{ plan_name }} {{ scan_motor }} {{ start }} {{ stop }} {{ num }} {{ time }}")
 
-_SCANS_WITHOUT_MOTORS = ['ct']
-_SCANS_WITH_MOTORS = ['ascan', 'dscan']
-_SPEC_SCAN_NAMES = _SCANS_WITH_MOTORS + _SCANS_WITHOUT_MOTORS
+_SCANS_WITHOUT_MOTORS = {'ct': 'count'}
```
Comment (danielballan) [inline suitcase/spec.py]: I see that this section has been refactored from collections to dicts. What's the reasoning behind this -- what is being mapped to what? I think we have count-like plans `['ct', 'count']` and scan-like plans `['ascan', 'scan', 'dscan', 'rel_scan']` and we need to recognize which group (if any) a given run represents. I don't think we need to _normalize_ the plan name -- we just need to classify as count-like or scan-like so that we know what columns to put in our table.

File: suitcase/nexus.py (hunk 10 above)
Comment (licode) [reply suitcase/nexus.py]: I will add fill=True at db.get_events.

File: suitcase/spec.py (hunk 11 above)
Comment (licode) [reply suitcase/spec.py]: Dict provides an easy way to map between spec name and bs plan name. Agree, we can figure out better groups for spec style scan as more names are added.

File: suitcase/spec.py [hunk 12]
```diff
@@ -1008,19 +1010,13 @@ def _get_acq_time(start, default_value=-1):
 
 def _get_plan_name(start):
     plan_name = start['plan_name']
-    if plan_name not in _BLUESKY_PLAN_NAMES:
-        warnings.warn(
```
Comment (danielballan) [inline suitcase/spec.py]: I think this part makes sense to keep; at least, I don't see why we'd want to change it as part of this PR updating suitcase to work with the new databroker API.

Comment (licode) [reply suitcase/spec.py]: OK. Let's still return Others, if there is no match.

File: suitcase/nexus.py (hunk 10 above)
Comment (jrmlhermitte) [reply suitcase/nexus.py]: yes, i think that looks good, please add `fill=True`.

**Comment by jrmlhermitte** [comment]
If this PR is too long, could we add the hdf5 fix separately? LIX also needs this. I added a dirty fix for them but I'd rather not keep it long. thanks :-)

File: suitcase/spec.py (hunk 12 above)
Comment (tacaswell) [reply suitcase/spec.py]: I'm ok dropping the warning (and talked about this with Li) in person.  The fix for the warning would be to teach this code about more blusky name -> spec name mapping which is not really something we want to do.

The motivation to switching back to 'Other' is not knowing if spec readers have a white-listed list of names and will choke if we provide names outside of that.

File: suitcase/tests/test_spec.py [hunk 13]
```diff
@@ -227,7 +214,7 @@ def test_insert_specfile(spec_filename, db_all):
     # Can only insert the spec scans whose type is in the _BLUESKY_PLAN_NAMES
     # list
     scans_expected_to_fail = [scan for scan in specfile if scan.scan_command
-                              not in spec._BLUESKY_PLAN_NAMES]
```
Comment (tacaswell) [inline suitcase/tests/test_spec.py]: This logic seems to be inverted from what it was before?

File: suitcase/nexus.py (hunk 10 above)
Comment (licode) [reply suitcase/nexus.py]: thanks. I only made change in the hdf file, forget to do it at nexus file.

File: suitcase/tests/test_spec.py (hunk 13 above)
Comment (licode) [reply suitcase/tests/test_spec.py]: _spec_scan_names is dictionary with keys from  old _BLUESKY_PLAN_NAMES, which is ['dscan', 'ascan', 'ct'].

**Comment by danielballan** [comment]
@jrmlhermitte Can you hoist that "dirty fix" into a PR on here?

**Comment by jrmlhermitte** [comment]
@danielballan the dirty fix was a personal fix.
What was happening is that currently suitcase calls `fill_event` which relies on the legacy `db`.
At LIX, they had to write a custom handler to handle all cases of their file format. Basically, I add this custom handler to the legacy `db` (from memory):
```py
from databroker import databroker as db2
db2.reg.register_handler("AD_CBF", CustomHandler)
```
we make sure to call it `db2` so as to not conflict with the existing `db`. Also, this `db2` I believe reads from `_legacy_config.yml` and not the same config that is used in the startup file but I have made sure they're the same. Small point but I thought I would point that out. I don't recommend PR'ing this.
I can ask Lin to update his profiles to github so we can keep track of this change.
(https://www.github.com/nsls-ii-lix/profile_processing)
